src/test/suite/connectionController.test.ts

```diff
@@ -5,20 +5,56 @@ import { before, after } from 'mocha';
 import ConnectionController, {
   DataServiceEventTypes
 } from '../../connectionController';
+import { StorageController, StorageVariables } from '../../storage';
+import { StorageScope, DefaultSavingLocations } from '../../storage/storageController';
 import { StatusView } from '../../views';
 
-const testDatabaseURI = 'mongodb://localhost';
+import { TestExtensionContext } from './stubs';
+
+const testDatabaseURI = 'mongodb://localhost:27018';
+const testDatabaseInstanceId = 'localhost:27018';
 const testDatabaseURI2WithTimeout =
   'mongodb://shouldfail?connectTimeoutMS=1000&serverSelectionTimeoutMS=1000';
 
 suite('Connection Controller Test Suite', () => {
   vscode.window.showInformationMessage('Starting tests...');
 
-  before(require('mongodb-runner/mocha/before'));
-  after(require('mongodb-runner/mocha/after'));
+  const mockExtensionContext = new TestExtensionContext();
+  const mockStorageController = new StorageController(mockExtensionContext);
+
+  before(async () => {
+    require('mongodb-runner/mocha/before');
+    // Disable the dialogue for prompting the user where to store the connection.
+    await vscode.workspace.getConfiguration('mdb.connectionSaving').update(
+      'hideOptionToChooseWhereToSaveNewConnections',
+      true
+    );
+    // Don't save connections on default.
+    await vscode.workspace.getConfiguration('mdb.connectionSaving').update(
+      'defaultConnectionSavingLocation',
+      DefaultSavingLocations['Session Only']
+    );
+  });
+  after(async () => {
+    require('mongodb-runner/mocha/after');
```

Comment:
fyi: you may want to move these runner mocha helpers to `pretest` and `posttest` package.json script hooks. They will not function as expected as you have them currently implemented.

Reply:
ah.... hah. Totally had these in and hadn't looked into what they were up to yet. ty - removed them since we already have them in our `pretest` & `posttest`.